libxrdp/libxrdp.c

```diff
@@ -1764,6 +1765,97 @@ libxrdp_send_session_info(struct xrdp_session *session, const char *data,
     return xrdp_rdp_send_session_info(rdp, data, data_bytes);
 }
 
+/*****************************************************************************/
+/*
+   Sanitise extended monitor attributes
+
+   The extended attributes are received from either
+   [MS-RDPEDISP] 2.2.2.2.1 (DISPLAYCONTROL_MONITOR_LAYOUT), or
+   [MS-RDPBCGR] 2.2.1.3.9.1 (TS_MONITOR_ATTRIBUTES)
+
+   @param monitor_layout struct containing extended attributes
+*/
+static void
+sanitise_extended_monitor_attributes(struct monitor_info *monitor_layout)
```

Comment:
I like this refactor, especially because it helps clean up and clarify what extended monitor attributes are!